docs/the-new-architecture/pillars-turbomodule.md

```diff
@@ -3,16 +3,699 @@ id: pillars-turbomodules
+
+```typescript title="NativeCalculator.ts"
+import type { TurboModule } from 'react-native/Libraries/TurboModule/RCTExport';
+import { TurboModuleRegistry } from 'react-native';
+
+export interface Spec extends TurboModule {
+  add(a: number, b: number): Promise<number>;
+}
+
+export default (TurboModuleRegistry.get<Spec>(
+  'RTNCalculator'
+) as Spec | null);
+```
+
+</TabItem>
+</Tabs>
+
+At the beginning of the spec files are the imports:
+
+- The `TurboModule` type, which defines the base interface for all TurboModules
+- The `TurboModuleRegistry` JavaScript module, which contains functions for loading TurboModules
+
+The second section of the file contains the interface specification for the TurboModule. In this case, the interface defines the `add` function which takes two numbers and returns a promise that resolves to a number. This interface type **must** be named `Spec` for a TurboModule.
+
+Finally, we invoke `TurboModuleRegistry.get`, passing the module's name, which will load the TurboModule if it's available. 
+
+:::caution
+We are writing JavaScript files importing types from libraries, without setting up a proper node module and installing its dependencies. Your IDE will not be able to resolve the import statements and you may see errors and warnings. This is expected and will not cause problems when you add the module to your app.
+:::
+
+## 3. Module Configuration
+
+Next, you need to add some configuration for [**Codegen**](/docs/pillars-codegen.md) and auto-linking.
+
+Some of these configuration files are shared between iOS and Android, while the others are platform-specific.
+
+### Shared
+
+The shared configuration is a `package.json` file that will be used by yarn when installing your module. Create the `package.json` file in the root of the `RTNCalculator` directory.
+
+```json title="package.json"
+{
+  "name": "rtn-calculator",
+  "version": "0.0.1",
+  "description": "Add numbers with TurboModules",
+  "react-native": "js/index",
+  "source": "js/index",
+  "files": [
+    "js",
+    "android",
+    "ios",
+    "rtn-calculator.podspec",
+    "!android/build",
+    "!ios/build",
+    "!**/__tests__",
+    "!**/__fixtures__",
+    "!**/__mocks__"
+  ],
+  "keywords": [
+    "react-native",
+    "ios",
+    "android"
+  ],
+  "repository": "https://github.com/<your_github_handle>/rtn-calculator",
+  "author": "<Your Name> <your_email@your_provider.com> (https://github.com/<your_github_handle>)",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/<your_github_handle>/rtn-calculator/issues"
+  },
+  "homepage": "https://github.com/<your_github_handle>/rtn-calculator#readme",
+  "devDependencies": {},
+  "peerDependencies": {
+    "react": "*",
+    "react-native": "*"
+  },
+  "codegenConfig": {
+    "libraries": [
+      {
+        "name": "RTNCalculatorSpec",
+        "type": "modules",
+        "jsSrcsDir": "js"
+      }
+    ]
+  }
+}
+```
+
+### iOS: Create the `podspec` file
```

Comment:
Point of interest: [The style guide recommends putting Android before iOS](https://github.com/facebook/react-native-website/blob/main/STYLEGUIDE.md#android-before-ios), but I don't think that's been consistently followed. Is that still something we'd like to keep?

Reply:
In this case, I would prefer local consistency over global consistency. IMHO let's keep the same ordering we followed in the other pages. We can re-sort them after we merged all of them 👍